### ISSUE TITLE
Can't bind Ctrl-j

### ISSUE THREAD

**Issue by willothy** [zellij-org/zellij#2679]
`zellij --version`: 0.37.2
`stty size`:  50 254
`uname -av` or `ver`(Windows): Linux ... 6.4.8-arch1-1 #1 SMP PREEMPT_DYNAMIC ... x86_64 GNU/Linux

`wezterm --version`: wezterm 20230717-111633-94ab0496

I'm attempting to setup smooth pane changes between Neovim, Wezterm and Zellij. My intended bindings use `Ctrl h/j/k/l` to move focus, and `Alt h/j/k/l` to resize. All of those work, except for `Ctrl j`, and I can't figure out why. I've disabled default keybinds in Zellij, ensured that the key isn't bound to anything in Wezterm, and still can't get it to work as expected.

To reproduce: 
- Set a normal mode binding `bind "Ctrl j" { MoveFocus "Down"; }`

**Comment by willothy** [comment]
Seems like a `termwiz` issue, see https://github.com/wez/wezterm/issues/2511

**Comment by p00f** [comment]
I can't bind Ctrl m either

**Comment by ssipos90** [comment]
I can't bind it either. Ctrl-j sends an enter. In tmux navigation is fine.

```
    normal clear-defaults=true {
        bind "Ctrl a" { SwitchToMode "Tmux"; }

        bind "Ctrl h" { MoveFocus "Left"; }
        bind "Ctrl j" { MoveFocus "Down"; }
        bind "Ctrl k" { MoveFocus "Up"; }
        bind "Ctrl l" { MoveFocus "Right"; }
    }
```

zellij 0.37.2 
alacritty 0.12.2 (9d9982df)
Manjaro Linux 23.0.0

**Comment by flinesse** [comment]
It seems that zellij mimics the behavior of many line editor's response to control sequences. As a quick test in addition to https://github.com/zellij-org/zellij/issues/2846#issuecomment-1760310459, I tried unbinding `Ctrl j` by giving `bindkey -r '^J'` in zsh and using the following keybinds with `clear-defaults=true` at the top level:
```
    shared_except "locked" {
        bind "Ctrl g"          { SwitchToMode "Locked"; }
        bind "Ctrl q"          { Detach; }
        bind "Ctrl Q"          { Quit; }
        bind "Ctrl ;"          { NewPane "Down"; }
        bind "Ctrl '"          { NewPane "Right"; }
        bind "Ctrl j"          { MoveFocus "Down"; }
        bind "Ctrl k"          { MoveFocus "Up"; }
        bind "Ctrl h"          { MoveFocusOrTab "Left"; }
        bind "Ctrl l"          { MoveFocusOrTab "Right"; }
        ...
    }
```
Doing `Ctrl j` now in zellij gives me a sound indicator by macOS that no program responded to this input.

There should probably be a caveat in the [allowable keybinds](https://zellij.dev/documentation/keybindings-keys) documentation that mentions this behavior for anyone looking to wholly customize zellij to their liking.

**Comment by swaits** [comment]
+1 on this on MacOS, wezterm, fish.

**Comment by fishel-feng** [comment]
+1

**Comment by Vinh-CHUC** [comment]
+1 on macos/tmux/zsh if that's relevant

**Comment by alankritjoshi** [comment]
I was losing my head over this today, thanks for reporting this!

Until this is fixed, I will be using [alt-centered-config](https://github.com/zellij-org/zellij/blob/main/example/alt-centered-config.kdl).

**Comment by cavanaug** [comment]
fwiw, as a vi user this defect is preventing me from really evaluating and switching to zellij

**Comment by hiasr** [comment]
At the moment, the issue lies with the way Zellij currently processes the key events. This is done through the [termwiz](https://github.com/wez/wezterm/tree/main/termwiz) crate provided by wezterm. They follow the ASCII specification where Ctrl+J is bound to line feed (\n) and Ctrl+m is bound to carriage return (\r), they do this by [hardcoding these inputs to enter](https://github.com/wez/wezterm/blob/cce0706b1f2a9e2d1f02c57f2d1cd367c91df1ae/termwiz/src/input.rs#L988). 

Binding Ctrl+j in tmux works because they first process the raw keybindings and send these to the terminal instead of converting them to enters. Most terminals nowadays use Ctrl+m as the enter command, you can try binding this in tmux and will see that your enter key now does the same as Ctrl+m. 

A possible solution would be to either handle these cases before parsing or request changes in upstream to be able to not parse control sequences (which would then need to be done in the client). Both solutions would be pretty hacky. I am mostly in favor of solution 1, with the note that it only happens if a keybinding is bound to e.g Ctrl+j, in order to prevent compatibility issues with other terminals which might use Ctrl+j as enter. I would be willing to implement this, but would first like feedback if the maintainers support this.

**Comment by imsnif** [comment]
Hi @hiasr - we preprocess such events in this function: https://github.com/zellij-org/zellij/blob/fca7a209d2c90ed8e2447f1c60ab630bc9455c13/zellij-utils/src/input/mod.rs#L61

If you can get it to work from there (possibly passing the client's keybindings to be able to see of `Ctrl j` was bound), I'd be willing to consider it. But I'll ask any such solution to be rigorously (manually) tested in all popular modern terminals to make sure we're not breaking anything when bound/unbound.

**Comment by hiasr** [comment]
This can be closed now that #3307 has been merged!
